Quick fix for infinite loop of item requests

ckostiw opened the pull request:
This is a quick fix that solves the immediate problem of a bad filter value causing an infinite loop of requests from a PaginatedTable, but which just makes the whole system a bit more complicated and difficult to understand. This could be merged if a better fix is going to take a long time to implement.

UlysseFG reviewed (approved):
Made a couple small additions to make ID search work for all sections.

ckostiw commented:
I added a new FILTER_TYPE.INPUT_OBJECT_ID to use explicitly for object id filters, and added validation on the input to block the user from setting a garbage value and generating a bad request error from the server.

UlysseFG commented:
> I added a new FILTER_TYPE.INPUT_OBJECT_ID to use explicitly for object id filters, and added validation on the input to block the user from setting a garbage value and generating a bad request error from the server.
FILTER_TYPE.INPUT_NUMBER was created for IDs ;) I am not sure it is helpful to have a filter type for each column eheh. The validation idea is nice and applies to any input_number.
